File: client/src/dto/gameState/TurnContextDto.ts
import { EMPTY_STRING } from "../../env";
import { GameContextDto, undefinedGameContextDto } from "./GameContextDto";
import { TurnProcessingResultDto } from "./TurnProcessingResultDto";

export interface TurnContextDto {
    currentPlayerId: string;
    isPlayer1Turn: boolean;
    remainingTimeInS: number;
    durationInS: number;
    notifyTurnChange: boolean;
    preMatchStart: boolean;
    gameContext: GameContextDto;
    newTurnProcessingInfo: TurnProcessingResultDto | null;
}

export const undefinedTurnContext: TurnContextDto = {
    currentPlayerId: EMPTY_STRING,
    isPlayer1Turn: false,
    remainingTimeInS: 0,
    durationInS: 0,
    notifyTurnChange: false,
    preMatchStart: false,
    gameContext: undefinedGameContextDto,
    newTurnProcessingInfo: null
};

File: server/dto/game_state/turn_context_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.game_state.game_context_dto import GameContextDto
from dto.game_state.turn_processing_result_dto import TurnProcessingResultDto


@dataclass
class TurnContextDto(BaseDto):
    """
    Contains information about the current turn context.
    Note : A turn context is meant to be sent to a single client,
    as it contains information that the other client should not be aware of.
    """

    currentPlayerId: str
    isPlayer1Turn: bool
    remainingTimeInS: int
    durationInS: int
    notifyTurnChange: bool
    gameContext: GameContextDto
    # None if not for a turn change
    newTurnProcessingInfo: TurnProcessingResultDto | None = None
